_posts/2024-02-22-refactoring-webgis-map-components.md

```diff
@@ -0,0 +1,94 @@
+---
+title: 'Refactoring web GIS map components'
+author: Janusz Lavrnja-Czapski
+categories:
+  - Development
+tags:
+  - javascript
+  - typescript
+  - best practices
+  - web gis
+---
+
+### The problem:
+
+In the context of the VENTURA VDR, we have recently received the go-ahead to create an open-source version of the project, which will be hosted on GitHub and open to future collaboration from the wider scientific/development community. This has prompted us to consider how we can get the codebase into a state where it is well-documented, organized, and ready for contributions.
```

Comment:
Maybe start with "BGS have participated as partners in the VENTURA research project [link] and created a prototype digital tool for collaborative early-stage strategic planning for future water management.  With agreement from the project partners and funding body, we are now creating an open source version of the project, etc"